src/middleware/find.js

```diff
@@ -1,18 +1,61 @@
 import React from 'react'
-let { TestUtils } = React.addons
+const { TestUtils } = React.addons
 
-export default function find(selector){
-  let elements, name
+const doFind = (root, selector) => {
   if (!(typeof selector === "string")) {
-    name = selector.name.toLowerCase()
-    elements = TestUtils.scryRenderedComponentsWithType(this.instance, selector)
-  } else if (selector.match(/\./)) {
-    selector = selector.replace(/\./, '')
-    elements = TestUtils.scryRenderedDOMComponentsWithClass(this.instance, selector)
+    return {
+      elements: TestUtils.scryRenderedComponentsWithType(root, selector),
+      name: selector.name.toLowerCase()
+    }
+  }
+
+  if (selector.match(/\./)) {
+    const className = selector.replace(/\./, '')
+    return {
+      elements: TestUtils.scryRenderedDOMComponentsWithClass(root, className),
+      name: className
+    }
+  }
+
+  return {
+    elements: TestUtils.scryRenderedDOMComponentsWithTag(root, selector),
+    name: selector
+  }
+}
+
+const flatten = array => {
+  return [].concat.apply([], array)
+}
+
+const findSingleOrMulti = (root, selector) => {
+  if (Array.isArray(root)) {
+    var matches = root.map(element => doFind(element, selector))
+    return {
+      elements: flatten(matches.map(match => match.elements)),
+      name: matches[0].name
+    }
+  }
+  return doFind(root, selector)
+}
+
+export default function find(selector, options) {
+  const {root, multi} = options || {}
+  const rootElement = root ? this.instance : this.root
+
+  const {elements, name} = findSingleOrMulti(rootElement, selector)
+
+  if (!elements || elements.length === 0)
+    throw new Error(`Could not find element "${selector}"`)
```

Comment:
I think it is necessary to throw. Because the element[selector] will be undefined That is good enough check. Is there more rationale in an throwing error??